src/sourcemap/CodeCoverage.zig

```diff
@@ -415,7 +477,7 @@ pub const ByteRangeMapping = struct {
                     executable_lines.set(line);
                     if (has_executed) {
                         lines_which_have_executed.set(line);
-                        lines_hits.ptr[line] += 1;
+                        line_hits_slice[line] += 1;
```

Comment:
@exoego writing to a many-item pointer (`[*]MyType`) directly is very much a footgun. You want to use a slice (`ptr[0..len]`), which has bounds checks.